### DISCUSSION TITLE
How to run METviewer on on NCAR's Casper

### DISCUSSION THREAD

**Discussion by YongxinFred** [dtcenter/METplus#2516] [Plot Generation] [answered]
I am running METplus on Casper to create verification statistics. I will use METviewer to create plots. I am just wondering if METviewer has been installed on Casper or any of the RAL machines and is ready to use. Thank you. Yongxin

**Comment by bikegeek** [comment, marked as answer]
You can check at this link to see where METviewer is installed: 
https://dtcenter.org/community-code/metplus/metplus-5-1-existing-builds

Unfortunately, it looks like the METviewer installation information hasn't been updated on this table.

There isn't an installation on Casper, but if you have access to any of the other RAL machines, such as mohawk, then you can start generating plots:

https://dtcenter.ucar.edu/met/metviewer/metviewer1.jsp


Here is a link to the METviewer User's Guide, if you don't already have access:
https://metviewer.readthedocs.io/en/latest/Users_Guide/index.html

The User's Guide has instructions for loading your data to the database

**Comment by YongxinFred** [comment]
Thank you so much for your quick response! I did see MET on hydro-c1. I will try to use METviewer on hydro-c1. Yongxin

**Reply by YongxinFred** [reply]
MET 11.0 was installed on hydro-c1 and I am wondering how to access MetViewer and the necessary database on hydro-c1. I didn't see any executables related to it. Thank you. Yongxin

**Comment by bikegeek** [comment]
Do you know who set up the METviewer instance on hydro-c1?  They should
have the database password and the URL to access it.  Can you access the
METviewer instance on 'mohawk'?

Here's the link: https://dtcenter.ucar.edu/met/metviewer/metviewer1.jsp

Here is the documentation that describes how to load your data into the
METviewer database (you'll need write privilege to the database):
https://metviewer.readthedocs.io/en/develop/Users_Guide/database_loading.html


**Command to load the database**

And here's how to use the database loader (mv_load.sh, which is in the METviewer source code) once you've set up your xml file following the instructions in the link above. The xml config file indicates the location of the database, the username, password, the MET stat output type (i.e. point_stat, ensemble_stat, etc.) the "group" where the data belongs, such as 'land-sea', 'hydro', etc.  whatever your group set up.  You can also create a new group.  

**Clone the METviewer code or cd to wherever it is located and enter the following:**

/path-to-METviewer-base-directory/METViewer/src/apps/METviewer/bin/mv_load.sh your_config.xml > & xml.log &

where the your_config.xml is your xml config file you set up to specify the data location, etc. You can name the log file to whatever you wish.  


Regards,
Minna

---------------
Minna Win
Pronouns: she/her
NSF NCAR
DTC & Research Applications Lab
Phone: 303-497-8423

*My work hours may not be your work hours. Please do not feel obliged to
reply to this email outside of your normal working hours.*
---------------

**Reply by YongxinFred** [reply]
Hi Minna,

     Thank you so much for your quick reply. I don't have access to "mohawk" but I will ask RAL Helpdesk if they could set up an account for me on "mohawk". If not, then I will ask for the database password and the URL to access it on "hydro-c1".

Yongxin

**Reply by YongxinFred** [reply]
Hi Minna,

     I just tried and I could access the METviewer instance on "mohawk" through the following URL that you sent to me earlier:

       https://dtcenter.ucar.edu/met/metviewer/metviewer1.jsp

     Does that mean that I don't need an account on "mohawk"?

Thanks,
Yongxin

**Reply by YongxinFred** [reply]
Hi Minna,

     I asked Tor to provide me the database password to the METviewer database on hydro-c1 so I could load my data to the database but Tor said that there was no mysql server running on hydro-c1. Tor asked me to ask you and MET team if mysql server needs to run and how to run this server. Thank you.

Yongxin
